src/coreclr/interpreter/compiler.cpp

```diff
@@ -1116,8 +1116,10 @@ void InterpCompiler::BuildEHInfo()
         int32_t handlerEndNativeOffset;
         GetNativeRangeForClause(clause.HandlerOffset, clause.HandlerOffset + clause.HandlerLength, &handlerStartNativeOffset, &handlerEndNativeOffset);
 
-        nativeClause.TryOffset = ConvertOffset(tryStartNativeOffset);
-        nativeClause.TryLength = ConvertOffset(tryEndNativeOffset);
+        // Interpreter-FIXME
+        const int hackOffsetToFixExceptionsThrownByFirstOpcodeInATryBlock = 1;
```

Comment:
It seems this would break other cases, can you please add an INTOP_NOP before the THROW_PNSE instead until I figure out what's wrong? I guess the problem is not in the instruction being the first in the try block, but rather in the instruction being the first one in the function. It sounds like a compensation "inherited" from native code handling is probably kicking in.

Reply:
Sure thing, I'll put a nop there instead. FWIW the instruction is not first in the function, but might be first in a "funclet"?